src/main/java/de/tum/in/www1/artemis/web/rest/QuizSubmissionResource.java

```diff
@@ -87,21 +88,27 @@ public QuizSubmissionResource(QuizExerciseRepository quizExerciseRepository, Qui
     }
 
     /**
+     * TODO: Decide if we want to use this endpoint for both submit and save. If so, we may want to use PUT instead of POST
+     * TODO: Don't trust the user submitted values
      * POST /exercises/:exerciseId/submissions/live : Submit a new quizSubmission for live mode.
      *
      * @param exerciseId     the id of the exercise for which to init a participation
      * @param quizSubmission the quizSubmission to submit
+     * @param submit         flag to determine if the submission should be submitted or saved
      * @return the ResponseEntity with status 200 (OK) and the Result as its body, or with status 4xx if the request is invalid
      */
     @PostMapping("exercises/{exerciseId}/submissions/live")
-    @EnforceAtLeastStudent
-    public ResponseEntity<QuizSubmission> submitForLiveMode(@PathVariable Long exerciseId, @Valid @RequestBody QuizSubmission quizSubmission) {
-        log.debug("REST request to submit QuizSubmission for live mode : {}", quizSubmission);
+    @EnforceAtLeastStudentInExercise
+    public ResponseEntity<QuizSubmission> saveOrSubmitForLiveMode(@PathVariable Long exerciseId, @Valid @RequestBody QuizSubmission quizSubmission,
+            @RequestParam(name = "submit", defaultValue = "false") boolean submit) {
+        log.debug("REST request to save or submit QuizSubmission for live mode : {}", quizSubmission);
         String userLogin = SecurityUtils.getCurrentUserLogin().orElseThrow();
         try {
-            // we set the submitted flag to true on the server side
-            quizSubmission.setSubmitted(true);
-            QuizSubmission updatedQuizSubmission = quizSubmissionService.saveSubmissionForLiveMode(exerciseId, quizSubmission, userLogin, true);
+            // we set the submitted flag on the server side
+            // TODO: For saving, there is a small change compared to QuizSubmissionWebsocketService::saveSubmission. The submitted flag is set here, but not in the websocket
```

Comment:
what is the difference? What are the implications?

Reply:
This comment refers to the deleted websocket way of submitting, where the endpoint assumed that `submitted` was `false` on the submission instead of setting it on the object.

As it's currently used, there's no meaningful difference